### ISSUE TITLE
pihole DNS entry incorrect, different from upstream DNS servers

### ISSUE THREAD

**Issue by Arindrew** [pi-hole/pi-hole#2937]
**In raising this issue, I confirm the following:** `{please fill the checkboxes, e.g: [X]}`

- [x] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [x] The issue I am reporting can be *replicated*.
- [x] The issue I am reporting isn't a duplicate (see [FAQs](https://github.com/pi-hole/pi-hole/wiki/FAQs), [closed issues](https://github.com/pi-hole/pi-hole/issues?utf8=%E2%9C%93&q=is%3Aissue%20is%3Aclosed%20), and [open issues](https://github.com/pi-hole/pi-hole/issues)).

**How familiar are you with the the source code relevant to this issue?:**

`1`

---
**Expected behaviour:**

`When performing a dig command with the pihole as the server, I expect it to reply with the same A record that it's upstream servers respond with.`

**Actual behaviour:**

`$ dig @1.1.1.1 plex.kandykastle.us +short`
`72.203.48.106`
`$ dig @192.168.10.10 plex.kandykastle.us +short`
`24.253.18.217`

**Debug token provided by [uploading `pihole -d` log](https://discourse.pi-hole.net/t/the-pihole-command-with-examples/738#debug):**

`https://tricorder.pi-hole.net/n22wxetpg1`

**Troubleshooting undertaken, and/or other relevant information:**

Ensured the hostname doesn't appear statically anywhere in /etc/ (/etc/hosts)
`sudo grep -r plex.kandykastle.us /etc`

**Comment by technicalpyro** [comment]
Likely the reason for this is caching and CDN related i imagine that URL resolves to both those IP addresses depending on geo-location

For example i get the same answer as you dig from one dot when i dig using my pi-hole.

**Comment by dschaper** [comment]
You have conditional forwarding set up for that domain. When you `dig` on 1.1.1.1 you are bypassing Pi-hole and asking the authoritative name server and you get the intended IP. When you dig against the Pi-hole it is asking the router what the IP address is supposed to be per your explicit configuration.

**Comment by Arindrew** [comment]
> You have conditional forwarding set up for that domain.

Thanks, fixed. I guess I misunderstood what that actually did. I found the static DNS entry for plex in my router and removed that.

I turned it off, created `/etc/hosts.kandykastle.us`, then created `/etc/dnsmasq.d/02-kandykastle.conf` with a single line of `addn-hosts=/etc/hosts.kandykastle.us` then a `pihole restartdns` and it all works exactly how it should.
